samples/sensor/fdc2x1x/src/main.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2020 arithmetics ITS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <sys/util.h>
+#include <drivers/sensor.h>
+
+#define CH_BUF_INIT(m)          {},
+
+K_SEM_DEFINE(sem, 0, 1);
+
+#ifdef CONFIG_FDC2X1X_TRIGGER
+static void trigger_handler(const struct device *dev,
+			    struct sensor_trigger *trigger)
+{
+	switch (trigger->type) {
+	case SENSOR_TRIG_DATA_READY:
+		if (sensor_sample_fetch(dev)) {
+			printk("Sample fetch error\n");
+			return;
+		}
+		k_sem_give(&sem);
+		break;
+	default:
+		printk("Unknown trigger\n");
+	}
+}
+#endif
+
+#ifdef CONFIG_PM_DEVICE
+static void pm_cb(const struct device *dev,
+		  int status,
+		  void *context,
+		  void *arg)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(arg);
+
+	switch (*(uint32_t *)context) {
+	case DEVICE_PM_ACTIVE_STATE:
+		printk("Enter ACTIVE_STATE ");
+		break;
+	case DEVICE_PM_LOW_POWER_STATE:
+		printk("Enter LOW_POWER_STATE ");
+		break;
+	case DEVICE_PM_OFF_STATE:
+		printk("Enter OFF_STATE ");
+		break;
+	}
+
+	if (status) {
+		printk("Fail\n");
+	} else {
+		printk("Success\n");
+	}
+}
+#endif
+
+void main(void)
+{
+	struct sensor_value ch_buf[] = {
+		DT_FOREACH_CHILD(DT_INST(0, ti_fdc2x1x), CH_BUF_INIT)
```

Comment:
@galak @pabigot I'm curious about what you think about the continued use of DT_INST() in sample applications.

I'm not really in favor of it because it means that users have to understand instance numbers, which are complicated and basically just for driver authors. Getting instance numbers to work in a sane way outside of a driver breaks down fast when there are multiple devices of the same compatible.

Reply:
I wanted to make the sample application work with both, the 2 channel and the 4 channel model of the sensor. I didn't know how else to get this information inside the sample. Should I instead make it only read 2 channels for both models?

Reply:
Let's keep it for now and revisit if we come up with a coherent policy in this regard.